### PR TITLE
Improves the "disable ERP" config

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Disables lewd underclothing, producing lewd reagents, seeing lewd quirks, seeing "Naked - Aroused", lewd loadout items, and seeing genital prefs.

## How This Contributes To The Skyrat Roleplay Experience
The config working as intended is good

## Changelog

:cl:
code: Improved the "disable ERP" configs
/:cl:

### CODE REVIEW

**Comment by RimiNosha** [comment]
This will cause merge conflicts for #16019, so folk are aware.

**Comment by twilightwanderer** [comment]
The project that promoted the "ERP shutdown" feature never got the Skyrat-based server up and running.

**Comment by Tastyfish** [comment]
> This will cause merge conflicts for #16019, so folk are aware.

Note that PR is test merged so this cannot be merged or test merged until that is concluded.

File: code/controllers/configuration/configuration.dm [hunk 1]
```diff
@@ -89,7 +89,10 @@
 	LoadPolicy()
 	LoadChatFilter()
 
-	populate_interaction_instances() //SKYRAT EDIT ADDITION
+	// SKYRAT ADDITION START
```
Comment (Tastyfish) [inline code/controllers/configuration/configuration.dm]:
```suggestion
	// SKYRAT EDIT ADDITION START
```

File: code/controllers/configuration/configuration.dm [hunk 2]
```diff
@@ -89,7 +89,10 @@
...
+	populate_interaction_instances()
+	remove_erp_things()
+	// SKYRAT ADDITION END
```
Comment (Tastyfish) [inline code/controllers/configuration/configuration.dm]:
```suggestion
	// SKYRAT EDIT ADDITION END
```

File: code/controllers/subsystem/processing/quirks.dm [hunk 3]
```diff
@@ -57,6 +57,11 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 		if(initial(quirk_type.abstract_parent_type) == type)
 			continue
 
+		// SKYRAT ADDITION START
```
Comment (Tastyfish) [inline code/controllers/subsystem/processing/quirks.dm]:
```suggestion
		// SKYRAT EDIT ADDITION START
```

File: code/controllers/subsystem/processing/quirks.dm [hunk 4]
```diff
@@ -57,6 +57,11 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
...
+		if(initial(quirk_type.erp_quirk) && CONFIG_GET(flag/disable_erp_preferences))
+			continue
+		// SKYRAT ADDITION END
```
Comment (Tastyfish) [inline code/controllers/subsystem/processing/quirks.dm]:
```suggestion
		// SKYRAT EDIT ADDITION END
```

File: code/controllers/subsystem/processing/reagents.dm [hunk 5]
```diff
@@ -17,6 +17,19 @@ PROCESSING_SUBSYSTEM_DEF(reagents)
 	GLOB.fake_reagent_blacklist = list(/datum/reagent/medicine/c2, /datum/reagent/medicine, /datum/reagent/reaction_agent)
 	//Build GLOB lists - see holder.dm
 	build_chemical_reactions_lists()
+
+	//	SKYRAT ADDITION START
```
Comment (Tastyfish) [inline code/controllers/subsystem/processing/reagents.dm]:
```suggestion
	// SKYRAT EDIT ADDITION START
```

File: code/controllers/subsystem/processing/reagents.dm [hunk 6]
```diff
@@ -17,6 +17,19 @@ PROCESSING_SUBSYSTEM_DEF(reagents)
...
+		for(var/reaction_path in GLOB.chemical_reactions_list)
+			var/datum/chemical_reaction/reaction_datum = GLOB.chemical_reactions_list[reaction_path]
+			if(!reaction_datum.erp_reaction)
+				continue
+			GLOB.chemical_reactions_list -= reaction_path
+			for(var/reaction in reaction_datum.required_reagents)
+				var/list/reaction_list = GLOB.chemical_reactions_list_reactant_index[reaction]
+				if(reaction_list)
+					reaction_list -= reaction_datum
+	// SKYRAT ADDITION END
```
Comment (Tastyfish) [inline code/controllers/subsystem/processing/reagents.dm]:
```suggestion
	// SKYRAT EDIT ADDITION END
```

File: code/modules/client/preferences.dm [hunk 7]
```diff
@@ -184,6 +183,13 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 /datum/preferences/ui_static_data(mob/user)
 	var/list/data = list()
 
+	// SKYRAT ADDITION START
```
Comment (Tastyfish) [inline code/modules/client/preferences.dm]:
```suggestion
	// SKYRAT EDIT ADDITION START
```

File: code/modules/client/preferences.dm [hunk 8]
```diff
@@ -184,6 +183,13 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+	if(CONFIG_GET(flag/disable_erp_preferences))
+		data["preview_options"] = list(PREVIEW_PREF_JOB, PREVIEW_PREF_LOADOUT, PREVIEW_PREF_UNDERWEAR, PREVIEW_PREF_NAKED)
+	else
+		data["preview_options"] = list(PREVIEW_PREF_JOB, PREVIEW_PREF_LOADOUT, PREVIEW_PREF_UNDERWEAR, PREVIEW_PREF_NAKED, PREVIEW_PREF_NAKED_AROUSED)
+	// SKYRAT ADDITION END
```
Comment (Tastyfish) [inline code/modules/client/preferences.dm]:
```suggestion
	// SKYRAT EDIT ADDITION END
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 9]
```diff
@@ -894,6 +894,11 @@
 			if(initial(quirk_type.abstract_parent_type) == type)
 				continue
 
+			// SKYRAT ADDITION START
```
Comment (Tastyfish) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
			// SKYRAT EDIT ADDITION START
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 10]
```diff
@@ -894,6 +894,11 @@
...
+			if(initial(quirk_type.erp_quirk) && CONFIG_GET(flag/disable_erp_preferences))
+				continue
+			// SKYRAT ADDITION END
```
Comment (Tastyfish) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
			// SKYRAT EDIT ADDITION END
```

**Comment by ORCACommander** [comment]
> The project that promoted the "ERP shutdown" feature never got the Skyrat-based server up and running.

I am not sure what you are getting at here?
Either you are being salty about something that is not gonna affect SR main
OR,
You are saying my downstream never got off the ground?

**Pull request merged**